go/listbasic.go

```diff
@@ -0,0 +1,153 @@
+		opts.lt = lt
+		return nil
+	}
+}
+
+// ListBasicLTEOption only lists keys less or equal to the given argument
+func ListBasicLTEOption(lte string) ListBasicOption {
+	return func(opts *listBasicOptionSet) error {
+		opts.lte = lte
+		return nil
+	}
+}
+
+// ListBasicMaxKeysOption limits the number of returned keys (default and maximum is 10000).
+func ListBasicMaxKeysOption(maxKeys int) ListBasicOption {
+	return func(opts *listBasicOptionSet) error {
+		if maxKeys < 0 || maxKeys > 10000 {
+			return fmt.Errorf("maxKeys=%d is out of the valid range [0, 10000]", maxKeys)
+		}
+		opts.maxKeys = maxKeys
+		return nil
+	}
+}
+
+// ListBasicNoKeysOption declares that keys are not needed in the
+// result entries and may be returned empty.
+//
+// Note: keys may still be returned until ARSN-438 is fixed.
+func ListBasicNoKeysOption() ListBasicOption {
+	return func(opts *listBasicOptionSet) error {
+		opts.noKeys = true
+		return nil
+	}
+}
+
+// ListBasicNoValuesOption declares that values are not needed in the
+// result entries and may be returned empty.
+//
+// Note: values may still be returned until ARSN-438 is fixed.
+func ListBasicNoValuesOption() ListBasicOption {
+	return func(opts *listBasicOptionSet) error {
+		opts.noValues = true
+		return nil
+	}
+}
+
+type ListBasicEntry struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type ListBasicResponse []ListBasicEntry
+
+type listBasicOptionSet struct {
+	gt       string
+	gte      string
+	lt       string
+	lte      string
+	maxKeys  int
+	noKeys   bool
+	noValues bool
+}
+
+func parseListBasicOptions(opts []ListBasicOption) (listBasicOptionSet, error) {
+	parsedOpts := listBasicOptionSet{
+		maxKeys: -1,
+	}
+	for _, opt := range opts {
+		err := opt(&parsedOpts)
+		if err != nil {
+			return parsedOpts, err
+		}
+	}
+	return parsedOpts, nil
+}
+
+func (client *BucketClient) ListBasic(ctx context.Context,
+	bucketName string, opts ...ListBasicOption) (*ListBasicResponse, error) {
+	resource := fmt.Sprintf("/default/bucket/%s", bucketName)
+	query := url.Values{}
+	query.Set("listingType", "Basic")
+
+	options, err := parseListBasicOptions(opts)
+	if err != nil {
+		return nil, &BucketClientError{
+			"ListBasic", "GET", client.Endpoint, resource, 0, "", err,
+		}
+	}
+	if options.gt != "" {
+		query.Set("gt", options.gt)
+	}
+	if options.gte != "" {
+		query.Set("gte", options.gte)
+	}
+	if options.lt != "" {
+		query.Set("lt", options.lt)
+	}
+	if options.lte != "" {
+		query.Set("lte", options.lte)
+	}
+	if options.maxKeys != -1 {
+		query.Set("maxKeys", strconv.Itoa(options.maxKeys))
+	}
+	if options.noKeys {
+		query.Set("keys", "false")
+	}
+	if options.noValues {
+		query.Set("values", "false")
+	}
+	resource += "?" + query.Encode()
+	responseBody, err := client.Request(ctx, "ListBasic", "GET", resource)
+	if err != nil {
+		return nil, err
+	}
+	var parsedResponse = new(ListBasicResponse)
+	jsonErr := json.Unmarshal(responseBody, parsedResponse)
```

Comment:
very nit picky: 
Here we do not need a new allocation, probably we can use a variable like this to avoid an extra allocation

```suggestion
	var parsedResponse ListBasicResponse
	jsonErr := json.Unmarshal(responseBody, &parsedResponse)
```